lib/domain.gi

```diff
@@ -244,12 +244,18 @@ InstallImmediateMethod( Enumerator,
 InstallMethod( AsList,
     "for a domain with stored domain generators",
     [ IsDomain and HasGeneratorsOfDomain ],
-    D -> DuplicateFreeList( GeneratorsOfDomain( D ) ) );
+    function( D )
+        if IsDuplicateFreeList( GeneratorsOfDomain( D ) ) then
+            return GeneratorsOfDomain( D );
+        else
+            return DuplicateFreeList( GeneratorsOfDomain( D ) );
```

Comment:
Thinking about this some more, I am not really sure anymore whether this is an "improvement", as the description of this PR claims: `AsList` is an attribute. So we are talking about a one-time cost here. Now, with the old code, we'd just call `DuplicateFreeList`, which performs an `O(n^2)` algorithm.
With this "improvement", we instead first run `IsDuplicateFreeList`, which also is `O(n^2)`. And if it fails, we still run `DuplicateFreeList`. So while there may be scenarios where this saves something (namely memory, but not performance), there are also scenarios where it results in a slow-down.

Are there any real-world applications that motivate this change?

Reply:
Oh, I'm completely fine with changing the condition to
`HasIsDuplicateFreeList( GeneratorsOfDomain( D ) ) and IsDuplicateFreeList( GeneratorsOfDomain( D ) )`.

Reply:
> Are there any real-world applications that motivate this change?
I want `AsList(Domain([1..3]))` to return the range `[1..3]`. At the moment gap returns the plain list `[1,2,3]`. As the range `[1..3]` knows that it is a `IsDuplicateFreeList` in this situation there should be no reason to copy it and create a new plain list.

Reply:
I changed the condition to
`HasIsDuplicateFreeList( GeneratorsOfDomain( D ) ) and IsDuplicateFreeList( GeneratorsOfDomain( D ) )`

Reply:
With `HasIsDuplicateFreeList` it is of course fine.